ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala

```diff
@@ -103,81 +105,84 @@ private[index] class IndexServiceImpl(
       transactionFilter: domain.TransactionFilter,
       verbose: Boolean,
   )(implicit loggingContext: LoggingContext): Source[GetTransactionsResponse, NotUsed] =
-    between(startExclusive, endInclusive) { (from, to) =>
-      from.foreach(offset =>
-        Spans.setCurrentSpanAttribute(SpanAttribute.OffsetFrom, offset.toHexString)
-      )
-      to.foreach(offset =>
-        Spans.setCurrentSpanAttribute(SpanAttribute.OffsetTo, offset.toHexString)
-      )
-      dispatcher()
-        .startingAt(
-          from.getOrElse(Offset.beforeBegin),
-          RangeSource { (startExclusive, endInclusive) =>
-            filterSource(transactionFilter, verbose) {
-              (templateFilter, eventProjectionProperties) =>
-                transactionsReader
-                  .getFlatTransactions(
-                    startExclusive,
-                    endInclusive,
-                    templateFilter,
-                    eventProjectionProperties,
-                  )
-            }
-          },
-          to,
+    withValidatedFilter(transactionFilter, packageMetadataView.current()) {
```

Comment:
Validation at this layer looks a bit off. There is the option of keeping this concern separated and performing this validation in `GrpcTransactionService` in the same manner we do it for the `ledgerEnd` (i.e. expose the packageMetadataView in the `IndexService`), together with the other arguments validation. Wdyt @nmarton-da @skisel-da ?

Reply:
An attempt I made is here:
https://github.com/digital-asset/daml/pull/14737